### ISSUE TITLE
Training starts with a nan

### ISSUE THREAD

**Issue by Bismuth209** [MasahideOkada/animate-anyone#3]
```
accelerate launch --mixed_precision="fp16" \
    train_1st_stage.py \
    --pretrained_dir "checkpoints" \
    --train_data_dir "/home/ubuntu/data/animate-anyone/TikTok_dataset_micro_view2" \
    --csv_path "/home/ubuntu/data/animate-anyone/TikTok_dataset_micro_view2.csv"  \
    --train_batch_size 64 --max_train_steps 30000 \
    --output_dir "animate-stage1.1" \
    --train_batch_size=8
```

I'm using this on a H100 GPU trying to overfit on TikTok data of 20 videos. The loss however is starting with a nan even for learning rate of 5e-6.

**Comment by MasahideOkada** [comment]
add `torch.autograd.set_detect_anomaly(True)` above the training loop, run the script again. and can you share the error output?
the training loop starts at [line 778](https://github.com/MasahideOkada/animate-anyone/blob/main/train_1st_stage.py#L778).

**Comment by kecol** [comment]
The issue might stem from how the convolutional layers in the PoseGuider are initialized, particularly when using ReLU as the activation function. There's a possibility that information passing through deeper blocks could exceed boundaries, leading to values like +inf or nan.

To address this, consider adding the following method to the PoseGuider class. This method employs He initialization for the weights and sets biases to zero. After removing the current calls to the gaussian_module, invoke this method once at the end of the __init__() method:
`
    def _initialize_weights(self):
        # Initialize weights with He initialization and zero out the biases
        for m in [self.conv_in, *self.blocks, self.conv_out]:
            if isinstance(m, nn.Conv2d):
                n = m.kernel_size[0] * m.kernel_size[1] * m.in_channels
                nn.init.normal_(m.weight, mean=0.0, std=np.sqrt(2. / n))
                if m.bias is not None:
                    nn.init.zeros_(m.bias)
                    
`

This adjustment should help ensure that the weights and biases of the convolutional layers are initialized in a way that prevents the information from going out of bounds during the deeper stages of the network

**Comment by MasahideOkada** [comment]
Thanks for your advice. I updated PoseGuider.
